activesupport/lib/active_support/duration.rb

```diff
@@ -312,6 +319,63 @@ def %(other)
         raise_type_error(other)
       end
     end
+    alias :modulo :%
+
+    def abs
+      Duration.new(value.abs, @parts.transform_values(&:abs), @variable)
+    end
+
+    def div(other)
+      if Scalar === other
+        Duration.new(value.div(other.value), @parts.transform_values { |number| number.div(other.value) }, @variable)
+      elsif Duration === other
+        value.div(other.value)
+      elsif Numeric === other
+        Duration.new(value.div(other), @parts.transform_values { |number| number.div(other) }, @variable)
+      else
+        raise_type_error(other)
+      end
+    end
+
+    def divmod(other)
+      [div(other), self % other)]
```

Comment:
Buildkite is failing. This either needs another left parenthesis or the right one removed.

```suggestion
      [div(other), self % other]
```

Reply:
I also can't help but wonder if this would be more efficient written as `self.to_i.divmod(other.to_i)`. Then you'd be using a C function.

Reply:
> Buildkite is failing. This either needs another left parenthesis or the right one removed.

Ack, thank you

> I also can't help but wonder if this would be more efficient written as `value.divmod(other.value)`. Then you'd be using a C function.

I'm not sure that's the same thing as this. The delegation will return something like `[Numeric, Numeric]` whereas how I have it currently will be `[Duration, Duration]` depending on the type of `other`.

Reply:
> Ack, thank you

No problem!

> I'm not sure that's the same thing as this. The delegation will return something like `[Numeric, Numeric]` whereas how I have it currently will be `[Duration, Duration]` depending on the type of `other`.

Ahh, I overlooked that. And both the `div` and `%` functions on Duration already use the Numeric ones. Cool. Nevermind me.